### PR TITLE
Add index reg. Module 3,4,5,6,7

### PR DESCRIPTION
**Pull request by Xintong-Li**
Comments:
1. In module 3 the word "volume" (as a higher dimensional generalization of plane) is highlighted, but there's no definition box associated to this term. I added the word "volume" to the index of terms (but not index of definitions).
2. Changed the entry "coefficient matrix" (appeared in Module 7) to "Matrix!coefficient matrix" and "System of linear equations!coefficient matrix".
3. The definition of "dot product" in definition.tex is referring to the algebraic definition (p.53). Thus, the definition entry referring to this is "Dot product!algebraic definition".
4.Pending review: entry "scalar product", "direction".
5.For some unknown reasons, "v comp" appears before "proj" in the index. (Perhaps because \Comp is before \Proj lexicographicly?)

### CODE REVIEW

File: book/common/definitions.tex [hunk 1]
```diff
@@ -72,8 +72,7 @@
 \end{SaveDefinition}
 
 \begin{SaveDefinition}[key=ZeroVector, title={Zero Vector}]
-		The \emph{zero vector}\index{Zero vector ($\vec 0$)}\index[definitions]{Zero vector ($\vec 0$)}, notated as
-		$\vec 0$\index[symbols]{$\vec 0$},
+		The \emph{zero vector}\index{Zero vector ($\vec 0$)}\index[definitions]{Zero vector ($\vec 0$)}, notated as $\vec 0$[\index[symbols]{$\vec 0$},
```
Comment (siefkenj) [inline book/common/definitions.tex:75]: Extra `[`

File: book/common/definitions.tex [hunk 2]
```diff
@@ -252,19 +249,19 @@
 
 \begin{SaveDefinition}[key=Orthogonal, title={Orthogonal}]
 	Two vectors $\vec u$ and $\vec v$ are
-	\emph{orthogonal} to each other if $\vec u\cdot \vec v=0$. The word orthogonal
+	\emph{orthogonal}\index{Orthogonality}\index[definitions]{Orthogonality} to each other if $\vec u\cdot \vec v=0$. The word orthogonal
```
Comment (siefkenj) [inline book/common/definitions.tex:252]: -> `Orthogonal`

File: book/common/definitions.tex [hunk 3]
```diff
@@ -285,14 +282,14 @@
...
-	v$, is the closest point in $X$ to $\vec v$.
+	v$, is the closest point in $X$ to $\vec v$.\index[definitions]{Projection}\index{Projection}\index[symbols]{$\Proj_{X}\vec v$}
 \end{SaveDefinition}
 
 \begin{SaveDefinition}[key=Component, title={Vector Components}]
 	Let $\vec u$ and $\vec v\neq \vec 0$ be vectors. The
 	\emph{vector component of $\vec u$ in the $\vec v$ direction}, written $\Comp_{\vec
 	v}\vec u$, is the vector in the direction of $\vec v$ so that
-	$\vec u-\Comp_{\vec v}\vec u$ is orthogonal to $\vec v$.
+	$\vec u-\Comp_{\vec v}\vec u$ is orthogonal to $\vec v$.\index{Vector component}\index[definitions]{Vector component}\index[symbols]{$\Comp_{\vec v}\vec u$}
```
Comment (siefkenj) [inline book/common/definitions.tex:292]: `\Vcomp` will also work

File: book/modules/module6.tex [hunk 4]
```diff
@@ -121,7 +121,7 @@
 As mentioned earlier, subspaces and spans are deeply connected.
 This connection is given by the following theorem.
 
-\begin{theorem}(Subspace-Span)
+\begin{theorem}(Subspace-Span)\index{Subspace!Subspace-Span Theorem}\index{Span!Subspace-Span Theorem}
```
Comment (siefkenj) [inline book/modules/module6.tex:124]: We don't need both of these.

File: book/modules/module7.tex [hunk 5]
```diff
@@ -20,7 +20,7 @@
 \[
 	\underbrace{\mat{1&2&-2\\2&1&-5\\1&-4&1}}_{A}\mat{x\\y\\z}=\mat{-15\\-21\\18}.
 \]
-The matrix $A$ on the left is called the \emph{coefficient matrix}\index{Coefficient matrix} because it
+The matrix $A$ on the left is called the \emph{coefficient matrix}\index{Matrix!coefficient matrix}\index[definitions]{Matrix!coefficient matrix}\index{System of linear equations!coefficient matrix}\index[definitions]{System of linear equations!coefficient matrix} because it
```
Comment (siefkenj) [inline book/modules/module7.tex:23]: Remove the `[definitions]` ones

**Comment by Xintong-Li** [comment]
Update: I fixed all previous comments, added index to Chapter 8, and finished indices for `definitions.tex`

Comments reg. "Representation of v in a basis B"
The notation of a column vector with a subscript cannot be properly rendered in the index (it pops an error). Thus, I did not include that in the index.

Comment reg. "ordered basis"
There's no definition of "ordered basis", and there's no "first time reference" of this phrase, so I did not add any index entry referring to this term.

File: book/common/definitions.tex [hunk 6]
```diff
@@ -392,18 +389,18 @@
 
 \begin{SaveDefinition}[key=OrientationofaBasis, title={Orientation of a Basis}]
 	The ordered basis $\mathcal B=\{\vec b_{1},\ldots,\vec b_{n}\}$ is
-	\emph{right-handed} or
-	\emph{positively oriented} if it can be continuously transformed to the
+	\emph{right-handed}\index[definitions]{Basis!right-handed}\index{Basis!right handed} or
```
Comment (siefkenj) [inline book/common/definitions.tex:392]: Keep the hyphen in both

File: book/common/definitions.tex [hunk 7]
```diff
@@ -533,7 +530,7 @@
 	key=Onetoone,
 	title={One-to-one}]
 		
-	Let $f:X\to Y$ be a function. We say $f$ is \emph{one-to-one} (or \emph{injective}) if
+	Let $f:X\to Y$ be a function. We say $f$ is \emph{one-to-one}\index[definitions]{One-to-one function}\index{One-to-one function} (or \emph{injective}) if
```
Comment (siefkenj) [inline book/common/definitions.tex:533]: Let's add `Injective function` as well.

File: book/common/definitions.tex [hunk 8]
```diff
@@ -542,71 +539,72 @@
 	title={Onto}]
 		
 	Let $f:X\to Y$ be a function.
-	We say $f$ is \emph{onto} (or \emph{surjective}) if every point in the codomain of $f$ gets mapped to.
+	We say $f$ is \emph{onto}\index{Onto function}\index[definitions]{Onto function} (or \emph{surjective}) if every point in the codomain of $f$ gets mapped to.
```
Comment (siefkenj) [inline book/common/definitions.tex:542]: Let's add `Surjective function`

File: book/common/definitions.tex [hunk 9]
```diff
@@ -542,71 +539,72 @@
...
 	That is $\Range(f)=Y$.
 \end{SaveDefinition}
 
 \begin{SaveDefinition}[
 	key=IdentityMatrix,
 	title={Identity Matrix}]
 	
-	An \emph{identity matrix} is a square matrix with ones on the diagonal
+	An \emph{identity matrix}\index{Matrix!identity}\index[definitions]{Matrix!identity} is a square matrix with ones on the diagonal
```
Comment (siefkenj) [inline book/common/definitions.tex:550]: `identity` -> `identity matrix`. We want to clarify that we're not talking about a "matrix identity" but the "identity matrix"

File: book/modules/module6.tex [hunk 10]
```diff
@@ -121,7 +121,7 @@
 As mentioned earlier, subspaces and spans are deeply connected.
 This connection is given by the following theorem.
 
-\begin{theorem}(Subspace-Span)
+\begin{theorem}(Subspace-Span)=
```
Comment (siefkenj) [inline book/modules/module6.tex:124]: Why the extra `=`? While you're at it, change the `(...)` to `[...]`

**Comment by Xintong-Li** [comment]
Fixed Module 6 and Definition

**Pull request merged**